components/_impl/workers/subprocess_rpc.py

```diff
@@ -126,9 +130,17 @@ def _loop(self):
             now = time.time()
             with self._loop_lock:
                 for w_fd, (timeout, start_time) in tuple(self._active_reads.items()):
-                    if now - start_time >= timeout and w_fd in self._active_reads:
-                        os.write(w_fd, _TIMEOUT)
-                        self.pop(w_fd)
+                    if timeout:
```

Comment:
I actually think we should unconditionally check zombie status, and then optionally check timeout. (Probably represent no timeout as something like `[None, -1, pid]`?) If we have a long timeout but a process dies immediately, it's a better user experience to fail immediately.